lib/livebook_web/live/home_live.ex

```diff
@@ -462,7 +463,20 @@ defmodule LivebookWeb.HomeLive do
     end
   end
 
-  defp determine_file(_params), do: Livebook.Config.local_filesystem_home()
+  defp determine_file(_params) do
+    case Ets.fetch(:file_system, "default_file_system") do
+      :error ->
+        Livebook.Config.local_filesystem_home()
+
+      {:ok, default} ->
+        {_id, file} =
+          Enum.find(Livebook.Settings.file_systems(), fn {id, _file} ->
+            id == default.file_system_id
+          end)
+
+        FileSystem.File.new(file)
+    end
+  end
```

Comment:
We will need the default filesystem in more places, specifically here:

https://github.com/livebook-dev/livebook/blob/3aa34393c54afc657f08397b9298182e66fcb79c/lib/livebook_web/live/session_live/persistence_live.ex#L177

So we should add some functions to settings, like `Settings.default_file_system()`, `Settings.default_file_system_home()`.